src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        try {
+            HibernateUtil.getFactory().inTransaction(session ->
+                    session.persist(movieSession));
+            return movieSession;
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't add movieSession " + movieSession, e);
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try {
+            return HibernateUtil.getFactory().fromSession(session ->
+                    Optional.ofNullable(session.get(MovieSession.class, id)));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get movieSession by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try {
+            return HibernateUtil.getFactory()
+                    .fromSession(session -> session.createQuery("FROM MovieSession ms "
+            + "WHERE ms.movie.id = :movieId AND DATE(showTime) = :date", MovieSession.class)
+                    .setParameter("movieId", movieId).setParameter("date", date).getResultList());
```

Comment:
would it make sense to extract this complex lambda to a separate method? 

`.fromSession(session -> findAvailableSessions(session, movideId, date)`

```
private List<MovieSession> findAvailableSessions(Session session, Long movideId, LocalDate date) {
// content from your lambda
}